addons/partner_autocomplete/models/res_company.py

```diff
@@ -12,3 +18,87 @@ class ResCompany(models.Model):
     def _inverse_partner_gid(self):
         for company in self:
             company.partner_id.partner_gid = company.partner_gid
+
+    def _enrich(self):
+        """
+        This method calls the partner autocomplete service from IAP to enrich the partner related fields
+        of the company.
+        """
+        self.ensure_one()
+
+        company_domain = self._get_company_domain()
+        if not company_domain:
+            return {'error': _("Could not extract company domain. Impossible to enrich the company.")}
+
+        company_data = self.env['res.partner'].enrich_company(company_domain, False, self.vat)
+        if company_data.get('error'):
+            return {'error': company_data.get('error_message', _('Unkown reason'))}
+
+        additional_data = company_data.pop('additional_info', False)
+
+        if company_data.get('logo') and not self.partner_id.image_1920:
+            company_data['image_1920'] = self._get_logo(company_data['logo'])
+
+        # Keep only truthy values that are not already set on the target partner
+        company_data = {field: value for field, value in company_data.items()
+                        if field in self.partner_id._fields and value and not self.partner_id[field]}
+
+        # extract country and state ids from results (because of res.partner._format_data_company)
```

Comment:
I would move that explanation to the docstring of "_enrich_extract_ids" then.
And explain here that you have to do it for both company data and children data.